overlord/devicestate/devicemgr.go

```diff
@@ -172,12 +175,79 @@ func (m *DeviceManager) ensureOperational() error {
 	return nil
 }
 
+// ensureSnaps makes sure that the snaps from seed.yaml get installed
+// with the matching assertions
+func (m *DeviceManager) ensureSeedYaml() error {
+	m.state.Lock()
+	defer m.state.Unlock()
+
+	// FIXME: enable on classic?
+	//
+	// Disable seed.yaml on classic for now. In the long run we want
+	// classic to have a seed parsing as well so that we can install
+	// snaps in a classic environment (LP: #1609903). However right
+	// now it is under heavy development so until the dust
+	// settles we disable it.
+	if release.OnClassic {
+		return nil
+	}
+
+	all, err := snapstate.All(m.state)
+	if err != nil {
+		return err
+	}
+	if len(all) > 0 {
+		return nil
+	}
+
+	for _, chg := range m.state.Changes() {
```

Comment:
do we need this now? afaict PopulateStateFromSeed will make the change once and commit and then never work again because of the flag